engine/consensus/dkg/messaging_engine.go

```diff
@@ -104,40 +111,57 @@ func (e *MessagingEngine) Process(_ network.Channel, originID flow.Identifier, e
 func (e *MessagingEngine) process(originID flow.Identifier, event interface{}) error {
 	switch v := event.(type) {
 	case *msg.DKGMessage:
+		// messages are forwarded async rather than sync, because otherwise the message queue
+		// might get full when it's slow to process DKG messages synchronously and impact
+		// block rate.
+		e.forwardInboundMessageAsync(originID, v)
+		return nil
+	default:
+		return fmt.Errorf("invalid event type (%T)", event)
```

Comment:
this should be a sentinel error, like `engine.InvalidInputError`, so we can treat it accordingly: 
* for external inputs, this is expected during normal operations 
* for internal inputs, this would be fatal 

```suggestion
		return engine.NewInvalidInputErrorf("expecting input with type msg.DKGMessage, but got %T", event)
```